.github/workflows/ci_cd.yml

```
@@ -822,12 +822,24 @@ jobs:
       - name: Run Playwright tests
         run: just frontend/run ${{ matrix.script }} --workers=2
 
+      - name: Check for changes after running tests with -u
+        id: check_snapshots_changed
+        run: |
+          # shellcheck disable=SC2086
+          if [ -n "$(git diff --binary --name-only)" ]; then
+            git add . && git diff --staged --binary | gzip -f > /tmp/snapshot_diff.patch.gz
+            echo "snapshots_changed=true" >> $GITHUB_OUTPUT
+            git diff --binary --name-only
+          else
+            echo "snapshots_changed=false" >> $GITHUB_OUTPUT
+          fi
+
       - uses: actions/upload-artifact@v4
-        if: failure()
+        if: failure() || steps.check_snapshots_changed.outputs.snapshots_changed == 'true'
         id: test-results
         with:
-          name: ${{ matrix.name }}_test_results
-          path: frontend/test-results
+          name: ${{ matrix.name }}_snapshot_diffs
+          path: /tmp/
```

Small nit/question: there's only a single diff file, so we don't need to pluralise the artefact name. But do we need to use the complete path to it, by chance?

```suggestion
          name: ${{ matrix.name }}_snapshot_diff
          path: /tmp/snapshot_diff.patch.gz
```

---

Good spot with the complete path. It should upload just the file and not the whole folder since we already have a folder(artifact) declared for it.

---

Suggested changes have been implemented